tests/integration_test/jwt_auth_integration_test/jwt_auth_integration_test.go

```diff
@@ -441,22 +396,89 @@ func TestFrontendAndBackendAuthHeaders(t *testing.T) {
+			method:                           "GET",
+			path:                             "/disableauthsettotrue/constant/disableauthsettotrue",
+			enableJwtPadForwardPayloadHeader: true,
+			headers: map[string]string{
+				"Authorization": "Bearer " + testdata.FakeCloudTokenSingleAudience3,
+			},
+			wantHeaders: map[string]string{
+				"X-Endpoint-API-UserInfo": testdata.FakeCloudTokenSingleAudienc3Payload + "==",
+			},
+		},
 	}
 
 	for _, tc := range testData {
-		url := fmt.Sprintf("http://%v:%v%v", platform.GetLoopbackAddress(), s.Ports().ListenerPort, tc.path)
-		resp, err := echo_client.DoWithHeaders(url, tc.method, "", tc.headers)
+		t.Run(tc.desc, func(t *testing.T) {
+			s := env.NewTestEnv(platform.TestFrontendAndBackendAuthHeaders, platform.EchoRemote)
+			s.OverrideAuthentication(&confpb.Authentication{
+				Rules: []*confpb.AuthenticationRule{
+					{
+						Selector: "1.echo_api_endpoints_cloudesf_testing_cloud_goog.dynamic_routing_BearertokenConstantAddress",
+						Requirements: []*confpb.AuthRequirement{
+							{
+								ProviderId: testdata.TestAuthProvider,
+								Audiences:  "ok_audience",
+							},
+						},
+					},
+					{
+						Selector: "1.echo_api_endpoints_cloudesf_testing_cloud_goog.dynamic_routing_DisableAuthSetToTrue",
+						Requirements: []*confpb.AuthRequirement{
+							{
+								ProviderId: testdata.TestAuthProvider,
+								Audiences:  "ok_audience",
+							},
+							{
+								ProviderId: testdata.GoogleServiceAccountProvider,
+								Audiences:  "need-pad",
+							},
+						},
+					},
+				},
+			})
+			s.OverrideMockMetadata(
+				map[string]string{
+					fmt.Sprintf("%v?format=standard&audience=https://%v/bearertoken/constant", util.IdentityTokenPath, platform.GetLoopbackAddress()): "ya29.BackendAuthToken",
+				}, 0)
 
-		if err != nil {
-			t.Fatalf("Test Desc(%s): %v", tc.desc, err)
-		}
+			defer s.TearDown(t)
+			flags := utils.CommonArgs()
+			if tc.enableJwtPadForwardPayloadHeader {
+				flags = append(flags, "--jwt_pad_forward_payload_header")
+			}
 
-		gotResp := string(resp)
-		for wantKey, wantValue := range tc.wantHeaders {
-			wantHeader := fmt.Sprintf(`"%v": "%v"`, wantKey, wantValue)
-			if !strings.Contains(gotResp, wantHeader) {
-				t.Fatalf("Test Desc(%s) failed,\n  got: %v\n want: %v", tc.desc, gotResp, tc.wantHeaders)
+			if err := s.Setup(flags); err != nil {
+				t.Fatalf("fail to setup test env, %v", err)
 			}
-		}
+			url := fmt.Sprintf("http://%v:%v%v", platform.GetLoopbackAddress(), s.Ports().ListenerPort, tc.path)
+			resp, err := echo_client.DoWithHeaders(url, tc.method, "", tc.headers)
+
+			if err != nil {
+				t.Fatalf("Test Desc(%s): %v", tc.desc, err)
+			}
+
+			gotResp := string(resp)
+			for wantKey, wantValue := range tc.wantHeaders {
+				wantHeader := fmt.Sprintf(`"%v": "%v"`, wantKey, wantValue)
+				if !strings.Contains(gotResp, wantHeader) {
+					t.Fatalf("Test Desc(%s) failed,\n  got: %v\n want: %v", tc.desc, gotResp, wantHeader)
```

Comment:
nit: This is not precise. Maybe "got response: %v. Want header %v in the response."

Reply:
Done.